packages/odie-client/src/data/use-get-support-interactions.ts

```diff
@@ -17,9 +17,20 @@ export const useGetSupportInteractions = (
 ) => {
 	const path = `?per_page=${ per_page }&page=${ page }&status=${ status }`;
 	const isTestMode = isTestModeEnvironment();
+	const { data: canConnectToZendesk } = useCanConnectToZendeskMessaging( enabled );
+	// Only fetch Zendesk interactions if the user can connect to Zendesk.
+	const shouldFetch = enabled && ( canConnectToZendesk || ! provider?.includes( 'zendesk' ) );
```

Comment:
Could you explain the `! provider?.includes( 'zendesk' )` bit, please?

Reply:
Yes, sometimes it's zendesk, sometimes it's zendesk-staging.

Reply:
When enabled is true:

<ul>
<li>
<p>shouldFetch will be <strong>false</strong> whenever <code inline="">enabled</code> is <code inline="">false</code>.</p>
</li>
<li>
<p>When <code inline="">enabled</code> is <code inline="">true</code>:</p>
<ul>
<li>
<p>If <code inline="">canConnectToZendesk</code> is <code inline="">true</code>, the result is <strong>true</strong> regardless of the provider.</p>
</li>
<li>
<p>If <code inline="">canConnectToZendesk</code> is <code inline="">false</code>, the result is <strong>true</strong> only when the provider string <strong>does not</strong> contain “zendesk” (i.e., <code inline="">odie</code>).</p>
</li>
</ul>
</li>
</ul></body></html>